The HIVDB Sequence Reads Interpretation Program accepts a codon frequency table that stores in the **CodFISH** format. The CodFISH format consists of five columns:

1. gene (`PR`, `RT`, or `IN`);
2. position;
3. total number of reads of this position;
4. codon nucleotide triplet; and
5. total number of reads of this codon.


## Examples

Following list CodFISH files from 8 studies with Illumina FASTQ sequence read files are available online (SRA or Zenodo).
You can download the zipball of either study, decompress the `.codfish` files into a local folder, and submit one or more
`.codfish` files to [HIVDB Interpretation Program](/hivdb/by-reads/) for analysis.

1. [Avila-RÃ­os S (2016)][avila-rios-2016-pubmed]: [257 CodFISH files (zipball)][avila-rios-2016-download]
2. [Dalmat RR (2018)][dalmat-2018-pubmed]: [112 CodFISH files (zipball)][dalmat-2018-download]
3. [Huber M (2016)][huber-2016-pubmed]: [34 CodFISH files (zipball)][huber-2016-download]
4. [Jair K (2019][jair-2019-pubmed]: [42 CodFISH files (zipball)][jair-2019-download]
5. [Moscona R (2017)][moscona-2017-pubmed]: [80 CodFISH files (zipball)][moscona-2017-download]
6. [Nguyen T (2018)][nguyen-2018-pubmed]: [134 CodFISH files (zipball)][nguyen-2018-download]
7. [Ode H (2015)][ode-2015-pubmed]: [92 CodFISH files (zipball)][ode-2015-download]
8. [Telele NF (2019)][telele-2019-pubmed]: [109 CodFISH files (zipball)][telele-2019-download]

[avila-rios-2016-pubmed]: https://www.ncbi.nlm.nih.gov/pubmed/27736898
[dalmat-2018-pubmed]: https://www.ncbi.nlm.nih.gov/pubmed/30305383
[huber-2016-pubmed]: https://www.ncbi.nlm.nih.gov/pubmed/27867045
[jair-2019-pubmed]: https://www.ncbi.nlm.nih.gov/pubmed/30964884
[moscona-2017-pubmed]: https://www.ncbi.nlm.nih.gov/pubmed/28799325
[nguyen-2018-pubmed]: https://www.ncbi.nlm.nih.gov/pubmed/29873733
[ode-2015-pubmed]: https://www.ncbi.nlm.nih.gov/pubmed/26617593
[telele-2019-pubmed]: https://www.ncbi.nlm.nih.gov/pubmed/29765082
[avila-rios-2016-download]: $$CMS_PREFIX$$/downloads/codfish-examples/avila-rios-2016-nicaragua-pmid27736898.zip
[dalmat-2018-download]: $$CMS_PREFIX$$/downloads/codfish-examples/dalmat-2018-pmid30305383.zip
[huber-2016-download]: $$CMS_PREFIX$$/downloads/codfish-examples/huber-2016-pmid27867045.zip
[jair-2019-download]: $$CMS_PREFIX$$/downloads/codfish-examples/jair-2019-pmid30964884.zip
[moscona-2017-download]: $$CMS_PREFIX$$/downloads/codfish-examples/moscona-2017-pmid28799325.zip
[nguyen-2018-download]: $$CMS_PREFIX$$/downloads/codfish-examples/nguyen-2018-pmid29873733.zip
[ode-2015-download]: $$CMS_PREFIX$$/downloads/codfish-examples/ode-2015-pmid26617593.zip
[telele-2019-download]: $$CMS_PREFIX$$/downloads/codfish-examples/telele-2019-pmid29765082.zip


## Create `.codfish` file from `.fastq` file

1. Install Docker CE (https://docs.docker.com/install/).

2. Download script:

   ```bash
   sudo curl -sL https://raw.githubusercontent.com/hivdb/codfish/master/bin/fastq2codfish-docker -o /usr/local/bin/fastq2codfish
   chmod +x /usr/local/bin/fastq2codfish
   ```

3. Use following command to process FASTQ files and generate CodFISH files.

   ```bash
   fastq2codfish /path/to/folders/containing/fastq/files
   ```

   The script will automatically find every file named with an extension of `.fastq`, align them to `.sam` file and then extract
   the codon freqency table into `.codfish` file.
   
   The above command is adequate for most case of both paired or unpaired FASTQ files generated by Illumina with the filename
   pattern look like `*_L001_R1_001.fastq.gz` and `*_L001_R1_002.fastq.gz`. However, if your FSTAQ files are in other naming
   convention, please read [advanced usages#change naming convention](#change.naming.convention).

Note: the `fastq2codfish` script can only be executed in an Unix-like system. If you are using Microsoft Windows 10,
you need to install the [Windows Subsystem for Linux](https://docs.microsoft.com/en-us/windows/wsl/install-win10) to
use this script.

### Advanced usages
#### Change naming convention
You can specify your FASTQ file naming convention by passing `-p <PATTERN>`, `-r <REPLACE>`, `-1 <PAIR1_SUFFIX>` and 
`-2 <PAIR2_SUFFIX>` parameters to `fastq2codfish`. Noted `-p` and `-r` are paired regular expression replacement.
For example, if your FASTQ files are downloaded from Sequence Reads Archive (SRA), their naming convention would be
`*_1.fastq.gz` and `*_2.fastq.gz`. Here is the command:

```bash
fastq2codfish -p '(.+)_[12]\.fastq\.gz$' -r '\1' -1 '_1.fastq.gz' -2 '_2.fastq.gz' /path/to/folders/containing/fastq/files
```

#### Specify concurrency rate
You can also specify concurrency rate (how many CPUs you want to use) by passing `-n <NTHREADS>` to `fastq2codfish`.
By default, the script will use fomula `FLOOR(TOTAL_CPUS * 19 / 20)` to calculate the `NTHREADS` value.

```bash
# Use at most 5 CPUs
fastq2codfish -n 5 /path/to/folders/containing/fastq/files
```

Note: if you are using Docker on Windows or MacOS, the upper limit of `NTHREADS` depends on the number of CPUs allocated
for the Docker software. Please check these general answers from Stackoverflow:

- Windows: https://stackoverflow.com/a/56583203/2644759
- MacOS: https://stackoverflow.com/a/39720010/2644759